gcloud-java-contrib/gcloud-java-nio/src/main/java/com/google/gcloud/storage/contrib/nio/CloudStorageFileSystemProvider.java

```diff
@@ -86,9 +86,13 @@ public CloudStorageFileSystemProvider() {
     this(storageOptions);
   }
 
-  private CloudStorageFileSystemProvider(@Nullable StorageOptions gcsStorageOptions) {
+  CloudStorageFileSystemProvider(@Nullable StorageOptions gcsStorageOptions) {
     if (gcsStorageOptions == null) {
-      this.storage = StorageOptions.defaultInstance().service();
+      if (storageOptions == null) {
+        this.storage = StorageOptions.defaultInstance().service();
+      } else {
+        this.storage = storageOptions.service();
```

Comment:
In prior comments @jart suggested that she will do something about `CloudStorageFileSystemProvide.setGCloudOptions` (hopefully remove it). I think it could be pretty confusing
when both options are provided as one would be ignored.